src/controller/python/chip-device-ctrl.py

```diff
@@ -719,8 +734,228 @@ def do_EOF(self, line):
+        else:
+            return __get_response_dict(status = StatusCodeEnum.SUCCESS)
+
+    except Exception as e:
+        return __get_response_dict(status = StatusCodeEnum.FAILED, error = str(e))
+
+def zcl_toggle_led(node_id: int, endpoint_id: Optional[int] = 1, group_id: Optional[int] = 0) -> Dict[str, Any] :
+    try:
+        __check_supported_os()
+        args = {}
+        err, res = device_manager.devCtrl.ZCLSend("OnOff", "Toggle", node_id, endpoint_id, group_id, args, blocking=True)
+        if err != 0:
+            return __get_response_dict(status = StatusCodeEnum.FAILED)
+        elif res != None:
+            return __get_response_dict(status = StatusCodeEnum.SUCCESS, result = str(res))
+        else:
+            return __get_response_dict(status = StatusCodeEnum.SUCCESS)
+    except Exception as e:
+        return __get_response_dict(status = StatusCodeEnum.FAILED, error = str(e))
+
+def zcl_add_thread_network(node_id: int, dataset: str, endpoint_id: Optional[int] = 1, group_id: Optional[int] = 0, breadcrumb: Optional[int] = 0, timeoutMs: Optional[int] = 1000) -> Dict[str, Any] :
+    try:
+        __check_supported_os()
+        args = {}
+        args['operationalDataset'] = bytes.fromhex(dataset)
+        args['breadcrumb'] = breadcrumb
+        args['timeoutMs'] = timeoutMs
+        err, res = device_manager.devCtrl.ZCLSend("NetworkCommissioning", "AddThreadNetwork", node_id, endpoint_id, group_id, args, blocking=True)
+        if err != 0:
+            return __get_response_dict(status = StatusCodeEnum.FAILED)
+        elif res != None:
+            return __get_response_dict(status = StatusCodeEnum.SUCCESS, result = str(res))
+        else:
+            return __get_response_dict(status = StatusCodeEnum.SUCCESS)
+
+    except Exception as e:
+        return __get_response_dict(status = StatusCodeEnum.FAILED, error = str(e))
+
+def zcl_enable_thread_network(node_id: int, network:str, endpoint_id: Optional[int] = 1, group_id: Optional[int] = 0, breadcrumb: Optional[int] = 0, timeoutMs: Optional[int] = 1000) -> Dict[str, Any]:
+    try:
+        __check_supported_os()
+        args = {}
+        args['networkID'] = bytes.fromhex(str(network))
+        args['breadcrumb'] = breadcrumb
+        args['timeoutMs'] = timeoutMs
+
+        err, res = device_manager.devCtrl.ZCLSend("NetworkCommissioning", "EnableNetwork", node_id, endpoint_id, group_id, args, blocking=True)
+        if err != 0:
+            return __get_response_dict(status = StatusCodeEnum.FAILED)
+        else:
+            return __get_response_dict(status = StatusCodeEnum.SUCCESS, result = str(res))
+    except Exception as e:
+        return __get_response_dict(status = StatusCodeEnum.FAILED, error = str(e)) 
+
+def zcl_enable_network(node_id: int, ssid:str, endpoint_id: Optional[int] = 1, group_id: Optional[int] = 0, breadcrumb: Optional[int] = 0, timeoutMs: Optional[int] = 1000) -> Dict[str, Any]:
```

Comment:
Same comment as above, can we club both enabling wifi and thread network into one function ?

Reply:
Done.